app/src/test/java/io/hstream/testing/Partition.java

```diff
@@ -43,6 +45,57 @@ public void setClient(HStreamClient client) {
     this.client = client;
   }
 
+  @Test
+  @Timeout(60)
+  void testWriteToMultiPartition() throws Throwable {
+    int ShardCnt = 5;
+    int threadCount = 10;
+    int count = 1000;
+    int keys = 16;
+    String streamName = randStream(client, ShardCnt);
+    BufferedProducer producer =
+        client.newBufferedProducer().stream(streamName)
+            .batchSetting(BatchSetting.newBuilder().recordCountLimit(100).ageLimit(10).build())
+            .build();
+    var produced = new HashMap<String, TestUtils.RecordsPair>();
+    assertExceptions(
+        runWithThreads(
+            threadCount,
+            () -> {
+              var pairs = produce(producer, 128, count, new RandomKeyGenerator(keys));
```

Comment:
if you mean `mergePairs(List<RecordsPair> pairs)`, I think we already have it? I add an `extend` function for `RecordsPair`, see:
https://github.com/hstreamdb/integration-tests/blob/a2ccf21cdf07f8aae6f8751a2d362b3bd23dd89d/app/src/test/java/io/hstream/testing/TestUtils.java#L593-L596

if you mean `mergePairs(List<HashMap<String, RecordsPair>> pairs)`, perhaps it would be better to provide a wrapped method that performs append concurrently and eventually returns a `HashMap<String, RecordsPair>`. :thinking:

Reply:
> if you mean mergePairs(List<HashMap<String, RecordsPair>> pairs), perhaps it would be better to provide a wrapped method that performs append concurrently and eventually returns a HashMap<String, RecordsPair>

Yes, that is great

Reply:
done